src/libAtomVM/term.h

```diff
@@ -258,6 +283,36 @@ static inline int term_is_binary(term t)
     return 0;
 }
 
+/**
+ * @brief Checks if a term is a binary
+ *
+ * @details Returns 1 if a term is a binary stored on the heap, otherwise 0.
+ * @param t the term that will be checked.
+ * @return 1 if check succedes, 0 otherwise.
+ */
+static inline int term_is_refc_binary(term t)
+{
+    /* boxed: 10 */
+    if (term_is_boxed(t)) {
+        const term *boxed_value = term_to_const_term_ptr(t);
+        int masked_value = boxed_value[0] & 0x3F;
+        switch (masked_value) {
```

Comment:
let's collapse this switch into an `if` `else` statement.

Reply:
Fixed.  (used `C ? A : B` syntax)